test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -140,39 +140,14 @@ func GetNonPerformancesWorkers(nodeSelectorLabels map[string]string) ([]corev1.N
 	return nonPerformanceWorkerNodes, err
 }
 
-// GetMachineConfigDaemonByNode returns the machine-config-daemon pod that runs on the specified node
-func GetMachineConfigDaemonByNode(node *corev1.Node) (*corev1.Pod, error) {
-	listOptions := &client.ListOptions{
-		Namespace:     testutils.NamespaceMachineConfigOperator,
-		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}),
-		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}),
-	}
-
-	mcds := &corev1.PodList{}
-	if err := testclient.Client.List(context.TODO(), mcds, listOptions); err != nil {
-		return nil, err
-	}
-
-	if len(mcds.Items) < 1 {
-		return nil, fmt.Errorf("failed to get machine-config-daemon pod for the node %q", node.Name)
-	}
-	return &mcds.Items[0], nil
-}
-
-// ExecCommandOnMachineConfigDaemon returns the output of the command execution on the machine-config-daemon pod that runs on the specified node
-func ExecCommandOnMachineConfigDaemon(ctx context.Context, node *corev1.Node, command []string) ([]byte, error) {
-	mcd, err := GetMachineConfigDaemonByNode(node)
-	if err != nil {
-		return nil, err
-	}
-	testlog.Infof("found mcd %s for node %s", mcd.Name, node.Name)
-
-	return testpods.WaitForPodOutput(ctx, testclient.K8sClient, mcd, command)
+// ExecCommand returns the output of the command execution as a []byte
+func ExecCommand(ctx context.Context, node *corev1.Node, command []string) ([]byte, error) {
+	return nodeInspector.ExecCommand(ctx, node, command)
 }
 
-// ExecCommandOnNode executes given command on given node and returns the result
-func ExecCommandOnNode(ctx context.Context, cmd []string, node *corev1.Node) (string, error) {
-	out, err := ExecCommandOnMachineConfigDaemon(ctx, node, cmd)
+// ExecCommandToString returns the output of the command execution in a string format
+func ExecCommandToString(ctx context.Context, cmd []string, node *corev1.Node) (string, error) {
+	out, err := ExecCommand(ctx, node, cmd)
```

Comment:
Shallow functions usually add very little value, and basically the main reason to have them is backward compatibility. What's the benefit of this function? Is it really helpful?

Reply:
@ffromani IIRC it's for backward compatibility, because some caller expects `string` and no `[]byte`

Reply:
Maybe a cleaner solution would be adding and `Options` argument to and merge both function, but this is out of the scope of this PR IMO

Reply:
What Talor said. also, I agree we should merge the functions, and do so in a separate PR.

Reply:
A middleground could be add a separate `ToString()` helper and use it wherever needed

Reply:
That sounds like a good idea and since it does not requires new struct definitions (like `Options`) I would recommend to implement that as part of this PR